virtual-display-driver-cli/src/client.rs

```diff
@@ -0,0 +1,113 @@
+use std::io::Write as _;
+
+use driver_ipc::Monitor;
+use eyre::Context;
+use win_pipes::{NamedPipeClientReader, NamedPipeClientWriter};
+
+pub struct Client {
+    reader: NamedPipeClientReader,
+    writer: NamedPipeClientWriter,
+}
+
+impl Client {
+    pub fn connect() -> eyre::Result<Self> {
+        let (reader, writer) = win_pipes::NamedPipeClientOptions::new("virtualdisplaydriver")
+            .wait()
+            .access_duplex()
+            .mode_message()
+            .create()?;
+
+        Ok(Self { reader, writer })
+    }
+
+    pub fn list(&mut self) -> eyre::Result<Vec<Monitor>> {
+        self.send_command(&driver_ipc::Command::RequestState)?;
+        let response = self.receive_command()?;
+
+        match response {
+            driver_ipc::Command::ReplyState(state) => Ok(state),
+            _ => eyre::bail!("received unexpected reply from driver pipe"),
+        }
+    }
+
+    pub fn get(&mut self, id: driver_ipc::Id) -> eyre::Result<Monitor> {
+        let monitors = self.list()?;
+        let monitor = monitors.into_iter().find(|monitor| monitor.id == id);
+
+        match monitor {
+            Some(monitor) => Ok(monitor),
+            None => eyre::bail!("no virtual monitor with ID {} found", id),
+        }
+    }
+
+    pub fn notify(&mut self, monitors: Vec<driver_ipc::Monitor>) -> eyre::Result<()> {
+        let command = driver_ipc::Command::DriverNotify(monitors);
+
+        self.send_command(&command)?;
+
+        Ok(())
+    }
+
+    pub fn remove(&mut self, ids: Vec<driver_ipc::Id>) -> eyre::Result<()> {
+        let command = driver_ipc::Command::DriverRemove(ids);
+
+        self.send_command(&command)?;
+
+        Ok(())
+    }
+
+    pub fn remove_all(&mut self) -> eyre::Result<()> {
+        let command = driver_ipc::Command::DriverRemoveAll;
+
+        self.send_command(&command)?;
+
+        Ok(())
+    }
+
+    pub fn new_id(&mut self, preferred_id: Option<driver_ipc::Id>) -> eyre::Result<driver_ipc::Id> {
+        let monitors = self.list()?;
```

Comment:
I think with how the code is written, if the `add` CLI command is called with an existing ID (without this check in place), it will effectively act just like `update`: it will send a single `notify` IPC command including just that one monitor with that ID, which will replace the existing monitor.

I'd need to do more testing to confirm this, but in either case, I'm fine with removing the uniqueness check since it couldn't break anything.

Reply:
_Marking as resolved, this will be handled in a follow-up_

Reply:
> I think with how the code is written, if the add CLI command is called with an existing ID (without this check in place), it will effectively act just like update: it will send a single notify IPC command including just that one monitor with that ID, which will replace the existing monitor.

That is correct!